linkerd/stack/src/router.rs

```diff
@@ -21,12 +21,19 @@ pub struct Router<T, N> {
 }
 
 impl<K, N> NewRouter<K, N> {
-    pub fn new(new_recgonize: K, inner: N) -> Self {
+    fn new(new_recgonize: K, inner: N) -> Self {
         Self {
             new_recgonize,
             inner,
         }
     }
+
+    pub fn layer(new_recgonize: K) -> impl layer::Layer<N, Service = Self> + Clone
```

Comment:
nit/tioli: _might_ be nice to have a comment on here explaining what the `new_recognize` parameter does?